packages/blockchain-wallet-v4-frontend/src/modals/ActiveRewards/DepositForm/index.tsx

```diff
@@ -0,0 +1,190 @@
+      key: Analytics.WALLET_ACTIVE_REWARDS_DEPOSIT_VIEWED,
+      properties: {
+        currency: coin
+      }
+    })
+    earnActions.setCoinDisplay({ isAmountDisplayedInCrypto: true })
+    earnActions.fetchEDDDepositLimits({ currency: walletCurrency })
+  }, [])
+
+  if (error)
+    return underSanctionsMessage ? (
+      <DataError onClick={handleRefresh} message={{ message: underSanctionsMessage }} />
+    ) : (
+      <DataError onClick={handleRefresh} />
+    )
+
+  if (!data || isLoading || isNotAsked) {
+    return <Loading />
+  }
+
+  const {
+    activeRewardsRates,
+    depositFee,
+    earnEDDStatus: { eddNeeded, eddPassed, eddSubmitted },
+    payment,
+    rates
+  }: RemoteType = data
+
+  const { coinfig } = window.coins[coin]
+  const currencySymbol = Exchange.getSymbol(walletCurrency) as string
+  const depositAmount = (values && values.depositAmount) || 0
+  const isCustodial =
+    values && values?.earnDepositAccount && values.earnDepositAccount.type === 'CUSTODIAL'
+  const maxDepositFiat = maxFiat(earnDepositLimits.maxFiat, walletCurrency)
+  const fromAccountType = isCustodial ? 'TRADING' : 'USERKEY'
+  const depositAmountFiat: number = amountToFiat(true, depositAmount, coin, walletCurrency, rates)
+  const depositAmountError =
+    formErrors.depositAmount &&
+    typeof formErrors.depositAmount === 'string' &&
+    formErrors.depositAmount
+  const isErc20 = !!coinfig.type.erc20Address
+  const insufficientEth =
+    payment &&
+    isErc20 &&
+    !!coinfig.type.erc20Address &&
+    // @ts-ignore
+    !payment.isSufficientEthForErc20
+  const isEDDRequired = eddNeeded && !eddPassed && !eddSubmitted
+
+  const handleFormSubmit = () => {
+    earnActions.submitDepositForm({ formName: FORM_NAME })
+
+    analyticsActions.trackEvent({
+      key: Analytics.WALLET_ACTIVE_REWARDS_DEPOSIT_TRANSFER_CLICKED,
+      properties: {
+        amount: depositAmount,
+        amount_usd: depositAmountFiat,
+        currency: coin,
+        type: fromAccountType
+      }
+    })
+  }
+
+  const handleAmountChanged = (e) => {
+    analyticsActions.trackEvent({
+      key: Analytics.ACTIVE_REWARDS_CLIENT_DEPOSIT_AMOUNT_ENTERED,
+      properties: {
+        amount: Number(e.target.value),
+        amount_currency: coin,
+        currency: walletCurrency,
+        from_account_type: fromAccountType,
+        input_amount: Number(values.depositAmount),
+        rate: Number(activeRewardsRates[coin])
+      }
+    })
+  }
+
+  const handleMaxAmountClicked = () => {
+    earnActions.handleTransferMaxAmountClick({
+      amount: displayCoin ? earnDepositLimits.maxCoin : earnDepositLimits.maxFiat,
+      formName: FORM_NAME
+    })
+
+    analyticsActions.trackEvent({
+      key: Analytics.ACTIVE_REWARDS_CLIENT_DEPOSIT_MAX_AMOUNT_CLICKED,
+      properties: {
+        amount_currency: coin,
+        currency: walletCurrency,
+        from_account_type: fromAccountType
+      }
+    })
+  }
+
+  const handleMinAmountClicked = () => () => {
```

Comment:
yes, do you see something wrong?

Reply:
I think Milan referred to it being a function within a function
`const handleMinAmountClicked = () => () => {}`, having twice parenthesis instead of 
`const handleMinAmountClicked = () => {}`

Reply:
fixed